### ISSUE TITLE
Merten FunkTaster CONNECT;  Merten 5071xx no trigger options for basic command set 0 and 255

### ISSUE THREAD

**Issue by JDF1** [zwave-js/node-zwave-js#3586]
### Is your problem within Home Assistant (Core or Z-Wave JS Integration)?

NO, my problem is NOT within Home Assistant or the ZWave JS integration

### Is your problem within ZWaveJS2MQTT?

NO, my problem is NOT within ZWaveJS2MQTT

### Checklist

- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there.

### Describe the bug

What causes the bug?
The Merten wireless Z-wave switch sent only basic command values.
These values are not selectable in the screen; automations -->trigger type: Device --> name device eg. "Merten wall controller" --> trigger drop down list

I expect a value like; 'subtype: 54-32-0 (currentValue)'
Or a read only entity under this z-wave device that detect the 0 and 255 basic value
![No_basic_commandClass_on-off](https://user-images.githubusercontent.com/42492548/138613276-36a1d08c-3870-4ca2-ab63-109f34400328.PNG)
s

### Device information

Manufacturer:  Merten
Model name: 5071xx and 506219

**Device info**
506219
by Merten
Firmware: 1.8
Z-Wave Info
Source: Z-Wave JS
Device ID: 54
Device Status: Asleep
Device Ready: Yes
Secure: No

### How are you using `node-zwave-js`?

- [X] `zwavejs2mqtt` Docker image (latest)
- [ ] `zwavejs2mqtt` Docker image (dev)
- [ ] `zwavejs2mqtt` Docker manually built _(please specify branches)_
- [ ] `ioBroker.zwave2` adapter _(please specify version)_
- [X] `HomeAssistant zwave_js` integration  _(please specify version)_
- [ ] `pkg`
- [ ] `node-red-contrib-zwave-js` _(please specify version, double click node to find out)_
- [ ] Manually built from GitHub _(please specify branch)_
- [ ] Other _(please describe)_

### Which branches or versions?

version:
`node-zwave-js` branch:
Z-Wave JS
Current version: 0.1.45 (Changelog)

Home Assistant 2021.10.6
Frontend version: 20211007.1 - latest
System Health
Version	core-2021.10.6
Installation Type	Home Assistant OS
Development	false
Supervisor	true
Docker	true
User	root
Virtual Environment	false
Python Version	3.9.7
Operating System Family	Linux
Operating System Version	5.10.70
CPU Architecture	x86_64
Timezone	Europe/Amsterdam

### Did you change anything?

yes (please describe)

### If yes, what did you change?

I upgrade form Z-Wave (deprecated) version to z-wave JS

### Did this work before?

Yes (please describe)

### If yes, where did it work?

In the Z-Wave (deprecated) 
I used the switch to fire events/scipts in home assistent with the automation:
Here I used the event type below:

trigger type: event
`entity_id: zwave.merten_transmitter_2_gang_switch
basic_level: 255`
action: a bunch of devices to turn on

and automation:

trigger type: event
`entity_id: zwave.merten_transmitter_2_gang_switch
basic_level: 0`
action: a bunch of devices to turn off

### Attach Driver Logfile

dump of the debug log when I press the buttons:
On command:

```
2021-10-24T21:02:17.990Z SERIAL « 0x01090004003b032001ff14                                            (11 bytes)
2021-10-24T21:02:17.993Z SERIAL » [ACK]                                                                   (0x06)
2021-10-24T21:02:17.995Z DRIVER « [Node 054] [REQ] [ApplicationCommand]
                                  └─[BasicCCSet]
                                      target value: 255
2021-10-24T21:02:17.998Z CNTRLR   [Node 054] treating BasicCC::Set as a report
2021-10-24T21:02:18.000Z CNTRLR   [Node 054] [~] [Basic] currentValue: 0 => 255                     [Endpoint 0]
```

Off command:

```
2021-10-24T21:02:20.965Z SERIAL « 0x01090004003b03200100eb                                            (11 bytes)
2021-10-24T21:02:20.967Z SERIAL » [ACK]                                                                   (0x06)
2021-10-24T21:02:20.969Z DRIVER « [Node 054] [REQ] [ApplicationCommand]
                                  └─[BasicCCSet]
                                      target value: 0
2021-10-24T21:02:20.972Z CNTRLR   [Node 054] treating BasicCC::Set as a report
2021-10-24T21:02:20.974Z CNTRLR   [Node 054] [~] [Basic] currentValue: 255 => 0                     [Endpoint 0]
```

**Comment by kpine** [comment]
The recommendation was to enable the `enableBasicSetMapping` compat flag for this device.

I also noticed the the configuration parameters are all wrong. See [OpenHAB](https://www.openhab.org/addons/bindings/zwave/thing.html?manufacturer=merten&file=506219_0_0.html) and [HomeSeer](http://tane.nl/shop/Merten/Merten_bpd.pdf) for the correct parameters, if there's interest in fixing them.

**Comment by AlCalzone** [comment]
Please re-interview the device and **attach** a driver log of that as a file.

**Comment by JDF1** [comment]
[z-waveJS Logs.zip](https://github.com/zwave-js/node-zwave-js/files/7412834/z-waveJS.Logs.zip)

**Comment by AlCalzone** [comment]
Hmm, I neither see Node 54 in there nor a complete interview. To be sure: re-interview **does not** mean removing and including again.

**Comment by JDF1** [comment]
Hi Alcalzone, first thank you for picking up this case. I see that I reinterviewed an other Merten switch in the log. I'm sorry for the confusion.  I reinterview Node 059.  name: "Merten wandzender". This switch is the same as node 54. Hope this helps.

**Comment by AlCalzone** [comment]
Unfortunately, that interview didn't get very far. You have to wake up sleeping nodes like this in order for an interview to go through.

**Comment by JDF1** [comment]
A I see. In this log file I saw "ZWAVE: Node 59: interview COMPLETED, all values are updated". I hope this is the right one. 
[zwavejs2mqttLogs2.zip](https://github.com/zwave-js/node-zwave-js/files/7417403/zwavejs2mqttLogs2.zip)

**Comment by AlCalzone** [comment]
Ok that looks good. The node does not support any actuator CCs except Basic CC, so the value updates for `currentValue` should be exposed to applications.
I suspect there's some filtering going on in HomeAssistant.

**Comment by kpine** [comment]
@JDF1 Please attach a [network dump](https://www.home-assistant.io/integrations/zwave_js/#get-a-dump-of-the-current-network-state) from HA.

**Comment by kpine** [comment]
> The recommendation was to enable the `enableBasicSetMapping` compat flag for this device.

Sorry I meant to say `treatBasicSetAsEvent`. 🤦🏻‍♂️

**Comment by AlCalzone** [comment]
As discussed, I'm fine with either the current functionality or with `treatBasicSetAsEvent` enabled. IMO it would still be good to see if the out-of-the-box behavior is working properly in HA.

**Comment by JDF1** [comment]
Here you go:
[zwave_js_dump.zip](https://github.com/zwave-js/node-zwave-js/files/7421650/zwave_js_dump.zip)

"nodeId": 59

**Comment by kpine** [comment]
> 
> 
> Here you go: [zwave_js_dump.zip](https://github.com/zwave-js/node-zwave-js/files/7421650/zwave_js_dump.zip)
> 
> "nodeId": 59

@AlCalzone According to the dump, no Basic values are being exposed to HA.

**Comment by AlCalzone** [comment]
Hmm... to double check that the filtering is happening in the driver, @JDF1 could you make another node dump, but this time from zwavejs2mqtt?
=> https://zwave-js.github.io/zwavejs2mqtt/#/troubleshooting/generating-logs?id=export-nodejson

**Comment by JDF1** [comment]
Yes, In the zip the the dump for node 59 and the all the nodes dump:
[Nodes_Dmp.zip](https://github.com/zwave-js/node-zwave-js/files/7425362/Nodes_Dmp.zip)

**Comment by AlCalzone** [comment]
Nevermind, I'm dumb... The device **controls** Basic CC, but it does not **support** it. So there are no values. The `treatBasicSetAsEvent` way is correct then.

**Comment by AlCalzone** [comment]
@zwave-js-bot add compat flag to 0x007a:0x0001:0x0004
```jsonc
// The device uses Basic Set to control other devices
treatBasicSetAsEvent: true
```

**Comment by JDF1** [comment]
Hi guys, I dont know if it is correct to reply to this post, but I want to say that the Merten double switch 506219 (0x0001:0x0004) works great after your fix. Many thanks!
I can now select the basic value as a trigger. (as seen in the picture below)

Is it also possible to apply the same fix for the single Merten switch  5071xx (0x0001:0x0002) 
Or should I open a new request for this?


![Basic-CC png](https://user-images.githubusercontent.com/42492548/141662478-e24bb929-191f-426c-ab37-0c7595a871d8.png)

**Comment by AlCalzone** [comment]
@zwave-js-bot add compat flag to 0x007a:0x0001:0x0002
```jsonc
// The device uses Basic Set to control other devices
treatBasicSetAsEvent: true
```

**Comment by AlCalzone** [comment]
@JDF1 I've opened #3708. Can you test [this file](https://raw.githubusercontent.com/zwave-js/node-zwave-js/d4a03ee90742bc4e583384251e0c4e510e91c1fe/packages/config/config/devices/0x007a/5071xx.json) to verify that it works?
See https://zwave-js.github.io/zwavejs2mqtt/#/device-support/custom-device-files for a description how. Don't forget to re-interview the device afterwards.

**Comment by JDF1** [comment]
I really would like to check it, but I can't find the /<app-root-folder>/store/config folder. I run Home assistant OS.  And if I understood it right, I can't access it very easy.

ha info:
arch: amd64
channel: stable
docker: 20.10.8
hassos: "6.6"
homeassistant: 2021.11.3
operating_system: Home Assistant OS 6.6
supervisor: 2021.10.8
supported: true
supported_arch:
- amd64
- i386

**Comment by AlCalzone** [comment]
Ok that makes it difficult. You'll have to wait for the change to be released then (likely in driver version 8.8.0).

**Comment by JDF1** [comment]
Yes, the single Merten switch works also after updating the driver to 8.8.0. 
Thanks!
